### DISCUSSION TITLE
use transition callbacks

### DISCUSSION THREAD

**Discussion by Manborough** [stimulus-use/stimulus-use#126] [Ideas]
I'd like to suggest that some callbacks be added into the use transition steps.
-  enterFrom
-  enterActive
-  enterTo
-  leaveFrom
-  leaveActive
-  leaveTo
Currently (at least as far as I can find) there's no ways to transition height and width properties without javascript.

So you could on enterActive to call this.
``` this.element.style.height = `${this.element.scrollHeight}px`; ```

and after enterTo do
``` this.element.style.height = ``; ```

Or something like that.

**Comment by adrienpoly** [comment]
Something like transitionStart and transitionEnd callbacks would be nice indeed
